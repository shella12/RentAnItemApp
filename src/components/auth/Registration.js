import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import { IoIosArrowBack } from 'react-icons/io';
import { Link } from 'react-router-dom';

class Registration extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      passwordConfirmation: '',
      error: '',
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    const { email, password, passwordConfirmation } = this.state;
    const { handleSuccessfulAuth } = this.props;

    axios
      .post(
        'https://renthousebackend.onrender.com/registrations',
        {
          user: {
            email,
            password,
            password_confirmation: passwordConfirmation,
          },
        },
        { withCredentials: true },
      )
      .then((response) => {
        if (response.data.status === 'created') {
          handleSuccessfulAuth(response.data);
        } else {
          this.setState({ error: 'error happened: please check email and password' });
        }
      })
      .catch((error) => {
        this.setState({
          error: error.message,
        });
      });

    event.preventDefault();
  }

  render() {
    const {
      email, password, passwordConfirmation, error,
    } = this.state;

    return (
      <div className="add-house-section">
        <Link to="/">
          <div className="flex-center btn-back">
            <IoIosArrowBack className="back-icon" />
            <span>Back</span>
          </div>
        </Link>
        <div className="backdrop column flex-center">
          <h1>Sign Up</h1>
          <p>
            Hello there! Sign up and start managing your system
          </p>
          <form onSubmit={this.handleSubmit} className="column">
            <input
              className="input-text"
              type="email"
              name="email"
              placeholder="Email"
              value={email}
              onChange={this.handleChange}
              required
            />

            <input
              className="input-text"
              type="password"
              name="password"
              placeholder="Password"
              value={password}
              onChange={this.handleChange}
              required
            />

            <input
              className="input-text"
              type="password"
              name="passwordConfirmation"
              placeholder="Password confirmation"
              value={passwordConfirmation}
              onChange={this.handleChange}
              required
            />
            <p>{error}</p>

            <button type="submit" className="btn">Register</button>
          </form>
        </div>
      </div>
    );
  }
}

Registration.propTypes = {
  handleSuccessfulAuth: PropTypes.func.isRequired,
};

export default Registration;
