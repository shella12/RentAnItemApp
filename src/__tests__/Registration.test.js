import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import Registration from '../components/auth/Registration';
import store from '../redux/configureStore';

it('renders Footer correctly', () => {
  let tree;
  act(() => {
    tree = renderer.create(
      <React.StrictMode>
        <Provider store={store}>
          <MemoryRouter initialEntries={['/register']}>
            <Registration handleSuccessfulAuth={() => {}} />
          </MemoryRouter>
        </Provider>
      </React.StrictMode>,
    );
  });
  expect(tree).toMatchSnapshot();
});
