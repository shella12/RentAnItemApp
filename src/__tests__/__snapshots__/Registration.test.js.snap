// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<div
  className="add-house-section"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="flex-center btn-back"
    >
      <svg
        className="back-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
      <span>
        Back
      </span>
    </div>
  </a>
  <div
    className="backdrop column flex-center"
  >
    <h1>
      Sign Up
    </h1>
    <p>
      Hello there! Sign up and start managing your system
    </p>
    <form
      className="column"
      onSubmit={[Function]}
    >
      <input
        className="input-text"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="input-text"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <input
        className="input-text"
        name="passwordConfirmation"
        onChange={[Function]}
        placeholder="Password confirmation"
        required={true}
        type="password"
        value=""
      />
      <p />
      <button
        className="btn"
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</div>
`;
